pubsub.go

```diff
@@ -527,6 +532,13 @@ func WithSeenMessagesTTL(ttl time.Duration) Option {
 	}
 }
 
+func WithAppSpecificRpcInspector(inspector func(peer.ID, *RPC) bool) Option {
```

Comment:
Also, lets make a type alias for the user inspector fun, returning error.

Reply:
https://github.com/libp2p/go-libp2p-pubsub/pull/509/commits/776d8592b871718e6f63228f5ce8f9c919f941ea